core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/OutputManager.scala

```diff
@@ -107,15 +108,12 @@ class OutputManager(
     )
   }
 
-  def emitEpochMarker(epochMarker: EpochMarker): Unit = {
-    // find the network output ports within the scope of the marker
-    val outputsWithinScope =
-      networkOutputBuffers.filter(out => epochMarker.scope.links.map(_.id).contains(out._1._1))
-    // flush all network buffers of this operator, emit epoch marker to network
-    outputsWithinScope.foreach(kv => {
-      kv._2.flush()
-      kv._2.addEpochMarker(epochMarker)
-    })
+  def flush(onlyFor: Option[Set[PhysicalLinkIdentity]] = None): Unit = {
+    val buffersToFlush = onlyFor match {
+      case Some(links) => networkOutputBuffers.filter(out => links.contains(out._1._1)).values
+      case None        => networkOutputBuffers.values
+    }
+    buffersToFlush.foreach(_.flush())
   }
```

Comment:
either rename the `onlyFor` or add some documentation about this parameter?

Reply:
added docs.